hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -2038,4 +2098,24 @@ object TestCOWDataSource {
       df.withColumn(c, when(col(c).isNotNull, col(c)).otherwise(lit(null)))
     }
   }
+
+  def provideParamsForKeyGenTest(): java.util.List[Arguments] = {
+    java.util.Arrays.asList(
+      Arguments.of(
+        "org.apache.hudi.keygen.MockUserProvidedKeyGenerator",
+        KeyGeneratorType.USER_PROVIDED.name()),
+      Arguments.of(
+        "",
```

Comment:
isnt' the 2nd case here expected to fail? 
i.e. setting type to `USER_PROVIDED`, but not setting any class. 

Also, where is the other test I asked for, 
i.e. setting the `class`, but not explicitly setting the `type`